logging/google/cloud/logging/logger.py

```diff
@@ -91,7 +95,8 @@ def batch(self, client=None):
 
     def _make_entry_resource(self, text=None, info=None, message=None,
                              labels=None, insert_id=None, severity=None,
-                             http_request=None, timestamp=None):
+                             http_request=None, timestamp=None,
+                             resource=_GLOBAL_RESOURCE):
```

Comment:
Does it make sense (in the context of the API) to make the resource default to `_GLOBAL_RESOURCE` instead of leaving it empty and letting the server assign the default resource? Does leaving it empty still make the server use the global resource or is it a required parameter?

If the server allows it to be optional, I strongly feel that we should not be baking / duplicated server-side behavior/defaults into the client.